src/main/java/cms/gov/madie/measure/services/TestCaseService.java

```diff
@@ -161,15 +171,22 @@ public TestCase updateTestCase(
       testCase.setCreatedAt(existing.getCreatedAt());
       testCase.setCreatedBy(existing.getCreatedBy());
       testCase.setResourceUri(existing.getResourceUri());
+      // assure patientId is not overwritten
+      testCase.setPatientId(existing.getPatientId());
       measure.getTestCases().remove(existing);
     } else {
       // still allowing upsert
       testCase.setId(ObjectId.get().toString());
       testCase.setCreatedAt(now);
       testCase.setCreatedBy(username);
+      testCase.setPatientId(UUID.randomUUID());
     }
 
     TestCase validatedTestCase = validateTestCaseAsResource(testCase, accessToken);
+    if (enforcePatientIdFeatureFlag
```

Comment:
Looks like if the feature flag is off, then TestCase object will still be saved with Patient Id and the json may be saved with no matching patient ID. Is this expected ?

Reply:
@RohitKandimalla I think that's what the feature flag is for.

Reply:
Thats interesting, I wonder why we need want to do one and not the other.